### PR TITLE
SQL: create..select revisited [closes #370]

### LINKED ISSUE #370: Revisit CREATE..SELECT
CREATE ... SELECT is defined as CREATE + INSERT. So in the case of
```sql
create or replace table t0(
y int,
st timestamp(6) generated always as row start,
en timestamp(6) generated always as row end,
period for system_time (st, en)
) with system versioning;
-- insert/delete/update t0
create table t3 with system versioning as select * from t0;
```
the new table `t3` should *not* have historical rows from `t0`. It should have the structure
```sql
create or replace table t3(
y int,
st timestamp(6) not null,
en timestamp(6) not null
) with system versioning;
```
that is, `st` and `en` are *normal* timestamp columns — which is exactly what you would see when doing `select * from t0`. It also has `sys_trx_start` and `sys_trx_end` versioning columns, but they're hidden — which is exactly what you'd see when creating a table `with system versioning` and without explicitly defined versioning columns.

### PR DESCRIPTION
**Pull request by midenok**
Please, review.

### CODE REVIEW

File: sql/handler.cc [hunk 1]
```diff
@@ -6791,116 +6815,137 @@ bool Vers_parse_info::check_and_fix_implicit(
...
+        *versioned_write= false;
       }
       continue;
     }
 
-    if ((f->versioning == Column_definition::VERSIONING_NOT_SET &&
-         !with_system_versioning) ||
-        f->versioning == Column_definition::WITHOUT_VERSIONING)
+    uint flags_left= VERS_SYSTEM_FIELD;
+    if (flags_left && (vers_info.is_start(*f) || vers_info.is_end(*f)))
```
Comment (kevgs) [inline sql/handler.cc]: `flag_left` can be replaced with `VERS_SYSTEM_FIELD` here.

File: sql/handler.cc [hunk 2]
```diff
@@ -6791,116 +6815,137 @@ bool Vers_parse_info::check_and_fix_implicit(
...
     {
-      f->flags|= VERS_UPDATE_UNVERSIONED_FLAG;
-    }
-  }
+      uint sys_flag= f->flags & flags_left;
```
Comment (kevgs) [inline sql/handler.cc]: `flag_left` can be replaced with `VERS_SYSTEM_FIELD` here.

File: sql/handler.cc [hunk 3]
```diff
@@ -6791,116 +6815,137 @@ bool Vers_parse_info::check_and_fix_implicit(
...
+      flags_left^= sys_flag;
```
Comment (kevgs) [inline sql/handler.cc]: It's a dead store. `flags_left` looks suspicious.

File: sql/handler.h [hunk 4]
```diff
@@ -1713,59 +1713,64 @@ struct Vers_parse_info
...
+  {
+    as_row.end= field_name;
+    system_time.end= field_name;
+  }
+  bool is_start(const char *name) const;
+  bool is_end(const char *name) const;
+  bool is_start(const Create_field &f) const;
+  bool is_end(const Create_field &f) const;
+  bool fix_implicit(THD *thd, Alter_info *alter_info, bool integer_fields, int *added= NULL);
+  bool any_sys_field_declared() const
```
Comment (kevgs) [inline sql/handler.h]: Name can be better. It's not only check for sys fields, but also `PERIOD FOR SYSTEM_TIME` parts.

File: sql/handler.cc (hunk 3 above)
Comment (midenok) [reply sql/handler.cc]: It's a typo. Should be:
```c++
flags_left-= sys_flag;
```
Thanks!

**Pull request merged**